tools/toolchains/arm.py

```diff
@@ -66,6 +66,9 @@ def __init__(self, target, notify=None, macros=None,
         self.asm = [main_cc] + self.flags['common'] + self.flags['asm']
         self.cc = [main_cc] + self.flags['common'] + self.flags['c']
         self.cppc = [main_cc] + self.flags['common'] + self.flags['c'] + self.flags['cxx']
+        if target.name in ["REALTEK_RTL8195AM"]:
```

Comment:
Unfortunately, ENUM size has to be 4 bytes for RTL8195. That's why there's target-specific modifications in arm.py/gcc.py/iar.py.

I have extended mbed's hook layer to accept target-specific options. I will send the PR for review. If that's accepted, these target-specific hacks can be removed.